external-providers/java-external-provider/pkg/java_external_provider/bldtool/tool.go

```diff
@@ -0,0 +1,98 @@
+package bldtool
+
+import (
+	"context"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/konveyor/analyzer-lsp/external-providers/java-external-provider/pkg/java_external_provider/dependency"
+	"github.com/konveyor/analyzer-lsp/external-providers/java-external-provider/pkg/java_external_provider/dependency/labels"
+	"github.com/konveyor/analyzer-lsp/provider"
+	"go.lsp.dev/uri"
+)
+
+// keys used in dep.Extras for extra information about a dep
+const (
+	artifactIdKey = "artifactId"
+	groupIdKey    = "groupId"
+	pomPathKey    = "pomPath"
+	baseDepKey    = "baseDep"
+)
+
+const (
+	maven  = "maven"
+	gradle = "gradle"
+)
+
+const (
+	gradleDepErr   = "gradleErr"
+	fallbackDepErr = "fallbackDepErr"
+)
+
+// TODO; Replace the mvn URI stuff in provider with the
+// a specific maven downloader tool.
+type Downloader interface {
+	Download(context.Context) (string, error)
+}
+
+type BuildTool interface {
+	GetDependencies(context.Context) (map[uri.URI][]provider.DepDAGItem, error)
+	UseCache() (bool, error)
+	GetCachedDepError(errorCached map[string]error) (error, bool)
+	GetLocalRepoPath() string
+	GetSourceFileLocation(string, string, string) (string, error)
+	GetResolver(string) (dependency.Resolver, error)
+	ShouldResolve() bool
```

Comment:
What does shouldResolve() do?

Reply:
Should Resolve delegates to the build tool on whether this should resolve or not. 

Basically, resolution MUST happen for binaries, but does not need to occur in the case of source analysis for a Maven or Gradle project that is source code. 

Maybe rename to MustResolve, and add that comment?